File: app/src/main/java/xyz/rodit/snapmod/features/friendsfeed/FeedSorter.java
package xyz.rodit.snapmod.features.friendsfeed;

import java.util.List;

import xyz.rodit.snapmod.features.Contextual;
import xyz.rodit.snapmod.features.FeatureContext;
import xyz.rodit.snapmod.mappings.FriendsFeedView;

public abstract class FeedSorter extends Contextual {

    public FeedSorter(FeatureContext context) {
        super(context);
    }

    abstract boolean shouldApply();

    abstract List<FriendsFeedView> sort(List<FriendsFeedView> items);
}

File: app/src/main/java/xyz/rodit/snapmod/features/friendsfeed/FeedSorter.kt
package xyz.rodit.snapmod.features.friendsfeed

import xyz.rodit.snapmod.features.Contextual
import xyz.rodit.snapmod.features.FeatureContext
import xyz.rodit.snapmod.mappings.FriendsFeedView

abstract class FeedSorter(context: FeatureContext) : Contextual(context) {

    abstract fun shouldApply(): Boolean

    abstract fun sort(items: MutableList<FriendsFeedView>): MutableList<FriendsFeedView>
}